### PR TITLE
🐛  Source Shopify: Fix Inventory Level Substream - Incremental Sync Flawed Filtering

### PR DESCRIPTION
**Pull request by cobobrien**
## What
The incremental sync for Shopify Inventory Level substreams pulls a fraction of the data that it should. This is due to the combination of an unsortable Parent Stream (Location) and the functionality fo the `filter_records_newer_than_state` method. Detailed explanation with examples in the issue: https://github.com/airbytehq/airbyte/issues/32817 

## How
Adds Incremental Inventory Levels by using the `updated_at_min` filter. Is far more performant than what existed previously due to no longer requiring the `filter_records_newer_than_state` functionality.

Previously incremental substreams pulls had to query all existing data for that resource without any filtering, and then filter out the older-than-state data after the fact in the connector runtime (via the `filter_records_newer_than_state` method). This way, only the newer-than-state data is pulled via the REST interface. 

It runs in a fraction of the time of the old implementation and will pull all qualifying records

## Recommended reading order
1. `source.py`

## 🚨 User Impact 🚨
All qualifying `Inventory Level` records will be pulled. This incremental stream sync will also be much more performant than it currently is.

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py [hunk 1]
```diff
@@ -832,11 +832,23 @@ class MetafieldLocations(MetafieldShopifySubstream):
...
-        location_id = stream_slice["location_id"]
-        return f"locations/{location_id}/{self.data_field}.json"
+        return f"{self.data_field}.json"
+
+    def request_params(self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs):
+        params = {"limit": self.limit}
+        if next_page_token:
+            params.update(**next_page_token)
+        else:
+            params[self.filter_field] = kwargs["stream_slice"]["location_id"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:846]:
```suggestion
            params[self.filter_field] = kwargs["stream_slice"][self.slice_key]
```

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py [hunk 2]
```diff
@@ -408,7 +408,7 @@ def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Ite
                 # avoid checking `deleted` records for substreams, a.k.a `Metafields` streams,
                 # since `deleted` records are not available, thus we avoid HTTP-400 errors.
                 if self.deleted_cursor_field not in record:
-                    yield {self.slice_key: record[self.nested_record]}
+                    yield {self.slice_key: record[self.nested_record]} | (stream_state or {})
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:411]: Can you explain the reason of returning stream_state here?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments thanks for the contribution @cobobrien

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py (hunk 2 above)
Comment (cobobrien) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:411]: I required access to the original `updated_at` date (from the state file passed in as an argument to the sync execution) in the `request_params` method. I couldn't find a better approach: 

1. The `stream_state` in the `request_params` seems to be always an empty dict
2. `self.get_updated_state` does no good here as we're looking for the value to use for the `updated_at_min` param
3. This connector does not have a `self.state` property

I'm eager to know of a better way of achieving this as it felt a bit hacky to me too. Thanks @marcosmarxm 🙌

**Comment by cobobrien** [comment]
> Left some comments thanks for the contribution @cobobrien

@marcosmarxm I have updated this PR. Could I please get a review?

**Comment by bazarnov** [comment]
@cobobrien I'll take a look.

**Comment by bazarnov** [comment]
@cobobrien Thank you for your time spent on the problem, currently, I don't feel this PR would be needed in the light of this one: https://github.com/airbytehq/airbyte/pull/32345 which will migrate the `InventoryLevels` from `REST` to `BULK` and will fix the `state` issues as well.  

@marcosmarxm Should we still merge this in with what is being said? Overall the PR looks good.

**Comment by bazarnov** [comment]
The fix has been merged @marcosmarxm FYI.

**Comment by marcosmarxm** [comment]
@cobobrien can you try latest version of the connector?

**Comment by marcosmarxm** [comment]
Closed as #32345 solves the issue.

**Pull request closed without merging**